staging/src/k8s.io/component-base/metrics/counter.go

```diff
@@ -298,7 +300,12 @@ func (v *CounterVec) ResetLabelAllowLists() {
 	v.LabelValueAllowLists = nil
 }
 
-// WithContext is a no-op.
+// WithContext is a no-op for now, users should still attach this to vectors for seamless future API upgrades (which rely on the context).
+// This is done to keep extensions (such as exemplars) on the counter and not its derivatives.
+// Note that there are no actual uses for this in the codebase except for chaining it with `WithLabelValues`, which makes no use of the context.
```

Comment:
This could still be breaking for downstream users that rely on the earlier `WithContext(...).ctx` in some way or the other, albeit no such use case was found in Kubernetes itself. I'm assuming general API stability cycles apply for that API, so we may want to push this out gradually.